src/pynxtools_raman_multiformat/reader.py

```diff
@@ -0,0 +1,242 @@
+        self.eln_data = parse_yml(
+            file_path,
+            convert_dict=CONVERT_DICT,
+            parent_key="/ENTRY[entry]",
+        )
+        #self.txt_line_skips = self.eln_data.get('/ENTRY[entry]/skip')
+
+        return {}
+
+    def handle_txt_file(self, filepath) -> Dict[str, Any]:
+        self.read_txt_file(filepath)
+        return {}
+
+
+
+
+
+
+
+
+
+    def get_attr(self, key: str, path: str) -> Any:
+        """
+        Get the metadata that was stored in the main file.
+        """
+        # This uses the path given from the config file (path = the part after "@attrs:")
+        # to get the corresponting value stored at in the eln_data.yaml file, at the 
+        # path value.
+        return self.eln_data.get(path)
+
+    def read_txt_file(self, filepath):
+        with open(filepath, "r") as file:
+            lines = file.readlines()
+
+        # Initialize dictionaries to hold header and data sections
+        header_dict = {}
+        data = []
+        line_count = 0
+        header_length = None
+
+        # Track current section
+        current_section = None
+
+
+
+        for line in lines:
+            line_count += 1
+            line = line.strip()  # Remove any leading/trailing whitespace
+            if line.startswith("[Header]"):
+                current_section = "header"
+                continue
+            elif line.startswith("[Data]"):
+                header_length = line_count + 2
+                current_section = "data"
+
+                continue
+
+            # Parse the header section
+            if current_section == "header" and "=" in line:
+                key, value = line.split("=", 1)
+                header_dict[key.strip()] = value.strip()
+
+            # Parse the data section
+            elif current_section == "data" and "," in line:
+                if line_count <= header_length:
+                    if line.startswith("[Header]"):
+                        print("We have to do something here")
+                if line_count > header_length:
+                    values = line.split(",")
+                    data.append([float(values[0].strip()), float(values[1].strip())])
+
+        #transform linewise read data to colum style data
+
+
+
+        # Transform: [[A, B], [C, D], [E, F]] into [[A, C, E], [B, D, F]]
+        data = [list(item) for item in zip(*data)]
+
+        data = np.transpose(data)
+
+        data_dict = {
+            "data/x_values": data[:, 0],
+            "data/y_values": data[:, 1]
+        }
+
+        self.txt_data = data_dict
+        self.txt_header = header_dict
+
+
+    def read_txt_columns(self, filepath):
+        data = np.loadtxt(filepath, delimiter=',')
+
+        # Convert the array into a dictionary with labeled columns
+        data_dict = {
+            "data/x_values": data[:, 0],
+            "data/y_values": data[:, 1]
+        }
+
+        A=self.transform_nm_to_wavenumber(532.1,data_dict["data/x_values"])

```

Comment:
Do you need this still?

Reply:
yea, wasnt needed. Just for testing purposes. Removed.